packages/jellyfish-api-core/__tests__/category/loan/placeAuctionBid.test.ts

```diff
@@ -0,0 +1,599 @@
+  expect(auctionsAfter[0].batches[0].collaterals).toStrictEqual(['5000.00000000@DFI', '0.50000000@BTC'])
+  expect(auctionsAfter[0].batches[0].loan).toStrictEqual('500.00399539@TSLA')
+
+  await tGroup.waitForSync()
+}
+
+describe('placeAuctionBid success', () => {
+  beforeEach(async () => {
+    await tGroup.start()
+    await alice.container.waitForWalletCoinbaseMaturity()
+    await setup()
+  })
+
+  afterEach(async () => {
+    await tGroup.stop()
+  })
+
+  it('should placeAuctionBid', async () => {
+    const bobColAccBefore = await bob.rpc.account.getAccount(bobColAddr)
+    expect(bobColAccBefore).toStrictEqual(['8900.00000000@DFI', '545.45454546@TSLA'])
+
+    const bobTSLAAccBefore = bobColAccBefore.length > 0
+      ? bobColAccBefore.find((amt: string) => amt.split('@')[1] === 'TSLA')
+      : undefined
+    const bobTSLAAmtBefore = bobTSLAAccBefore !== undefined ? Number(bobTSLAAccBefore.split('@')[0]) : 0
+
+    const txid = await bob.rpc.loan.placeAuctionBid({
+      vaultId: bobVaultId,
+      index: 0,
+      from: bobColAddr,
+      amount: '526@TSLA' // (500.00399539 * 5%) + 500 = 525.00419516, min first bid is 525.00419516
+    })
+    expect(typeof txid).toStrictEqual('string')
+    await bob.container.generate(1)
+    await tGroup.waitForSync()
+
+    const bobColAccAfter = await bob.rpc.account.getAccount(bobColAddr)
+    expect(bobColAccAfter).toStrictEqual(['8900.00000000@DFI', '19.45454546@TSLA'])
+
+    const bobTSLAAccAfter = bobColAccAfter.length > 0
+      ? bobColAccAfter.find((amt: string) => amt.split('@')[1] === 'TSLA')
+      : undefined
+    const bobTSLAAmtAfter = bobTSLAAccAfter !== undefined ? Number(bobTSLAAccAfter.split('@')[0]) : 0
+
+    expect(bobTSLAAmtBefore - bobTSLAAmtAfter).toStrictEqual(526) // 545.45454546 - 19.45454546 = 526
+
+    const aliceColAccBefore = await alice.rpc.account.getAccount(aliceColAddr)
+    expect(aliceColAccBefore).toStrictEqual(['30000.00000000@DFI', '29999.00000000@BTC', '10000.00000000@TSLA'])
+
+    // test second round placeAuctionBid
+    await alice.rpc.loan.placeAuctionBid({
+      vaultId: bobVaultId,
+      index: 0,
+      from: aliceColAddr,
+      amount: '535@TSLA' // (526 * 1%) + 525 = 531.26
+    })
+    await alice.generate(1)
+    await tGroup.waitForSync()
+
+    // end the auction and alice win the bid
+    await bob.generate(36)
+
+    const auctionsAfter = await bob.container.call('listauctions')
+    expect(auctionsAfter).toStrictEqual([
+      {
+        vaultId: bobVaultId,
+        state: 'inLiquidation',
+        loanSchemeId: 'scheme',
+        ownerAddress: bobVaultAddr,
+        liquidationPenalty: 5,
+        liquidationHeight: 205,
+        batchCount: 1,
+        batches: [
+          {
+            index: 0,
+            collaterals: ['5000.00000000@DFI', '0.50000000@BTC'],
+            loan: '500.00428078@TSLA' // 535 / 1.05, https://github.com/DeFiCh/pinkpaper/tree/main/loan#collateral-auction
```

Comment:
I have less clarity on how this changed from initial `500.00399539@TSLA`. clearly `535 / 1.05 != 500.00428078`
from the pink paper link, the recovered should be paid back to the vault and the panelty(5%) should be burned. and theres more about if auctions yeild extra tokens. question!

- if the first auction batch yeilds more tokens as in this example, does that mean the next batch amount has any effect?

Reply:
Adding yesterday's comment again.
would be interesting to know how it looks like in the vault if auctions yield extra tokens.

Reply:
its dangling long time ago.. lemme refresh a bit..